integration_test/ops_agent_test.go

```diff
@@ -1831,7 +1838,15 @@ func fetchPID(ctx context.Context, logger *log.Logger, vm *gce.VM, processName s
 	if gce.IsWindows(vm.Platform) {
 		cmd = fmt.Sprintf("Get-Process -Name '%s' | Select-Object -Property Id | Format-Wide", processName)
 	} else {
-		cmd = "sudo pgrep " + processName
+		// pgrep has a limit of 15 characters to lookup processes
+		// using -f uses the full command line for lookup
+		// using the pattern "[p]rocessName" avoids matching itself
```

Comment:
From the link you posted:

The running pgrep or pkill process will never report itself as a match.

I Ithink this means you don't need to do the "[p]rocessName" thing. That thing is useful when you are doing something like `ps aux | grep [p]rocessName`, but pgrep/pkill is different from piping the output to grep.

Reply:
@martijnvans Yes, you're right `pgrep/pkill` won't report itself as match. I think I've should've been more explicit in the comment (I will update it). It doesn't match itself, it matches the `ssh` session that runs the command, so connection is lost (`exit status 255`).

To give more detail, I was getting two errors that drove me to this solution :
- `fetchPID` and `terminateProcess` were not working since the new service name `google_cloud_ops_agent_diagnostics` is longer that 15 characters, so I tried adding `-f` which matches with the full command line.
- When running `pkill -f` (in `terminateProcess`) the test was throwing `exit status 255` (see https://github.com/GoogleCloudPlatform/ops-agent/pull/852/commits/1eb7da43ba80c69d234d05e7eec59756a681d5d9) since it was matching the full command line of the `ssh` session too. So I added the `[p]rocessName` to fix that.